internal/tasks/fleet_selector_test.go

```diff
@@ -106,14 +106,29 @@ var _ = Describe("FleetSelector", func() {
 			Expect(device.Metadata.Owner).To(BeNil())
 		})
 
-		It("Fleet selector removed no overlap", func() {
+		It("Nil fleet selector matches all", func() {
 			testutil.CreateTestFleet(ctx, fleetStore, orgId, "fleet", nil, nil)
-			testutil.CreateTestDevice(ctx, deviceStore, orgId, "device", util.StrToPtr("Fleet/fleet"), nil, &map[string]string{"key": "value"})
+			testutil.CreateTestDevice(ctx, deviceStore, orgId, "nillabels", util.StrToPtr("Fleet/fleet"), nil, nil)
+			testutil.CreateTestDevice(ctx, deviceStore, orgId, "emptylabels", util.StrToPtr("Fleet/fleet"), nil, &map[string]string{})
+			testutil.CreateTestDevice(ctx, deviceStore, orgId, "device1", util.StrToPtr("Fleet/fleet"), nil, &map[string]string{"key1": "value1"})
+			testutil.CreateTestDevice(ctx, deviceStore, orgId, "device2", util.StrToPtr("Fleet/fleet"), nil, &map[string]string{"key2": "value2"})
 			err := logic.FleetSelectorUpdatedNoOverlapping(ctx)
 			Expect(err).ToNot(HaveOccurred())
-			device, err := deviceStore.Get(ctx, orgId, "device")
+
+			listParams := store.ListParams{Limit: 0}
+			devices, err := deviceStore.List(ctx, orgId, listParams)
+			Expect(err).ToNot(HaveOccurred())
+			Expect(len(devices.Items)).To(Equal(4))
+			for _, device := range devices.Items {
+				Expect(*device.Metadata.Owner).To(Equal("Fleet/fleet"))
+			}
+
+			callback := store.FleetStoreCallback(func(before *model.Fleet, after *model.Fleet) {})
+			err = fleetStore.Delete(ctx, orgId, callback, "fleet")
```

Comment:
```suggestion
			err = fleetStore.Delete(ctx, orgId, callback, "fleet")
```

I would suggest we move deletion to a separate test

It("should delete a fleet will nil selectors without errors")

so if something goes wrong with deletion, we have  more granular test output

Reply:
Done